### ISSUE TITLE
[Usage]: Running the libero_goal_ppo_openpi_pi05 configuration, but both the reward and success rate are decreasing.

### ISSUE THREAD

**Issue by 1847540790** [RLinf/RLinf#291]
### Problem description

My training images are as follows. Could you please explain why this is happening?

![Image](https://github.com/user-attachments/assets/6f265afc-12e4-4ee0-b219-fec09b55b337)

This is my configuration file.
defaults:
  - env/train: libero_goal
  - env/eval: libero_goal
  - override hydra/job_logging: stdout

hydra:
  run:
    dir: .
  output_subdir: null
  searchpath:
    - file://${oc.env:EMBODIED_PATH}/config/

cluster:
  num_nodes: 1
  component_placement:
    # actor,env,rollout: 0
    actor: 2,5-7
    env: 2,5-7
    rollout: 2,5-7

runner:
  task_type: embodied
  logger:
    log_path: "../results"
    project_name: rlinf
    experiment_name: "test_openpi"
    logger_backends: ["tensorboard"] # wandb, swanlab

  max_epochs: 1000
  max_steps: -1

  val_check_interval: -1
  save_interval: 5
  seq_length: 4096
  max_prompt_length: 50

algorithm:
  auto_reset: False
  ignore_terminations: False
  use_fixed_reset_state_ids: True
  normalize_advantages: True
  kl_penalty: kl  # how to estimate kl divergence: kl or kl_penalty
  group_size: 1
  max_episode_steps: 320
  use_ordered_reset_state_ids: False
  reward_coef: 1.0
  n_chunk_steps: 64
  n_eval_chunk_steps: 64
  num_group_envs: 4 #64
  rollout_epoch: 8 #8
  reward_type: chunk_level
  logprob_type: chunk_level
  entropy_type: token_level
  update_epoch: 4

  adv_type: gae
  loss_type: actor_critic
  loss_agg_func: "token-mean" 
  kl_beta: 0.0
  entropy_bonus: 0
  clip_ratio_high: 0.2
  clip_ratio_low: 0.2
  clip_ratio_c: 3.0
  value_clip: 0.2
  huber_delta: 10.0

  gamma: 0.99
  gae_lambda: 0.95

  filter_rewards: False
  rewards_lower_bound: 0.1
  rewards_upper_bound: 0.9

  sampling_params:
    use_greedy: False
    temperature_train: 1.0
    temperature_eval: 0.6
    top_k: 50
    top_p: 1.0
    repetition_penalty: 1.0
    add_BOS: False

  length_params:
    max_new_token: null
    max_length: 1024
    min_length: 1

env:
  group_name: "EnvGroup"
  channel:
    name: "env_buffer_list"
    queue_name: "obs_buffer"
    queue_size: 0
  enable_offload: False

rollout:
  group_name: "RolloutGroup"
  channel:
    name: ${env.channel.name}
    queue_name: "action_buffer"
    queue_size: 0
  mode: "colocate"
  generation_backend: "huggingface"
  model_dir: "/workspace/RLinf/model/openpi"
  recompute_logprobs: False
  unnorm_key: libero_10
  enable_offload: False
  pipeline_stage_num: 1

actor:
  group_name: "ActorGroup"
  channel:
    name: ${env.channel.name}
    queue_name: "replay_buffer"
    queue_size: 0
  training_backend: "fsdp"
  checkpoint_load_path: "/workspace/RLinf/model/openpi"
  checkpoint_save_path: "../results"
  micro_batch_size: 8 #128
  global_batch_size: 128 # maximum : global batch size = micro batch size * actor num_processes.
  seed: 42
  enable_offload: False

  model:
    model_name: "openpi"
    precision: null
    num_action_chunks: 5 # interface for the env
    action_dim: 7
    is_lora: True
    lora_rank: 32
    gradient_checkpointing: False
    use_wrist_image: True
    use_proprio: True
    num_steps: 5
    sharding_strategy: "no_shard"
    add_value_head: True
    # openpi specific parameters
    openpi:
      noise_level: 0.5
      action_chunk: ${actor.model.num_action_chunks}
      num_steps: ${actor.model.num_steps}
      train_expert_only: True
      action_env_dim: ${actor.model.action_dim}
      noise_method: "flow_sde"
      add_value_head: ${actor.model.add_value_head}
      pi05: True
      value_after_vlm: True
      value_vlm_mode: "mean_token"

  optim:
    lr: 5.0e-6
    value_lr: 1.0e-4
    adam_beta1: 0.9
    adam_beta2: 0.95
    adam_eps: 1.0e-05
    clip_grad: 1.0
    critic_warmup_steps: 0

reward:
  use_reward_model: False

critic:
  use_critic_model: False

### Before submitting a new issue...

- [x] Have you checked relevant issues, FAQs (https://rlinf.readthedocs.io/en/latest/rst_source/faq.html), or asked the chatbot at the top right corner of the [documentation page](https://rlinf.readthedocs.io).

**Comment by LiuZhihao2022** [comment]
Hi! We noticed two critical changes in your configuration:

**1. `num_group_envs` is too small**

You changed:

```
num_group_envs: 64 → 4
```

Using too few parallel environments leads to **poor gradient estimation**, which will reduce training performance.  
We recommend increasing `num_group_envs` back to a larger value.

---

**2. Using LoRA (`is_lora: True`) requires safer hyperparameters**

You have set `is_lora: True`. When LoRA is enabled, VLM finetuning becomes more sensitive to hyperparameters.  
Please use **more conservative settings**, such as a lower learning rate (see https://arxiv.org/abs/2510.25889).

**Recommendation**

- Increase `num_group_envs`
- If using LoRA, reduce the LR and tune other hyperparameters accordingly. You can also choose to disable LoRA.

These adjustments should help improve training stability.

**Comment by ZZJ-Asher** [comment]
Thank you very much. I will try again to add the number of environments and see the effect after removing LoRA. I really appreciate your open-sourcing of RLinf.Thanks again.
